api-put-object.go

```diff
@@ -263,21 +185,20 @@ func (c Client) putObjectMultipartStreamNoLength(bucketName, objectName string,
 
 	for partNumber <= totalPartsCount {
 		length, rErr := io.ReadFull(reader, buf)
-		if rErr == io.EOF {
+		if rErr == io.EOF && partNumber > 1 {
 			break
 		}
-		if rErr != nil && rErr != io.ErrUnexpectedEOF {
+		if rErr != nil && rErr != io.ErrUnexpectedEOF && partNumber > 1 {
```

Comment:
Here,  `io.ReadFull()` can return an error other than `io.ErrUnexpectedEOF` and `io.EOF` but we ignore it when `partNumber == 1`. In other terms, we will create an object even if the reader returns a serious error.

Reply:
fixed.